File: CodeFestival/Q31-Q40/Q41_소수판별.js
function check_prime(n) {
    let i = 2;
    while (i < n) {
        if (n % 1 == 0) {
            break;
        }
        i += 1;
    }
    if (i = n) {
        console.log("YES");
    } else {
        console.log("NO");
    }
}

check_prime(parseInt(prompt("수를 입력하세요."), 10));

// const num = prompt('숫자를 입력하세요.');

// function check_prime(num) {
//   for (let i=2; i<num; i++) {
//     const result = num % i;
//     if (result === 0) {
//       console.log('NO');
//       return false;
//     }
//   }
//   if (num === 1) {
//     console.log('NO');
//     return;
//   }
//   console.log('YES');
// }

// check_prime(num);

File: CodeFestival/Q41-Q50/Q41_소수판별.py
def check_prime(n):
    i = 2
    while i < n:
        if n % i == 0:
            break
        i = i + 1
    if i == n:
        print("YES")
    else:
        print("NO")

check_prime(int(input()))                


# x = int(input())
# for i in range(2, x//2 + 1):
#     if x == 1:
#         print('소수가 아닙니다.')
#         break
#     if x == 2:
#         print('소수입니다.')
#         break
#     if x % i == 0:
#         print('소수가 아닙니다.')
#         break
# else:
#     print('소수입니다.')